### PR TITLE
TLP-52

### PR DESCRIPTION
**Pull request by nepnep83**
Selenium testing for preferred page / spike

### CODE REVIEW

File: requirements_dev.in [hunk 1]
```diff
@@ -6,4 +6,5 @@ pep8-naming==0.13.3
 pip-tools==6.13.0
 pur==7.1.0
 pytest-cov==4.0.0
-safety==2.3.5
\ No newline at end of file
+safety==2.3.5
+selenium==4.10.0
```
Comment (Aidan23Smith) [inline requirements_dev.in:10]: New line

File: tests/features/features.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#    Given a person
+#    And their name is Aidan Smith
+#    And they are 25
+#    Then a person called Aidan Smith exists
+#    And their age is 25
+#
+#    When they change their name to Tom Powell
+#    And their age to 18
+#    Then a person called Tom Powell exists
+#    And their age is 18
```
Comment (Aidan23Smith) [inline tests/features/features.feature:13]: Don't need to upload this

File: tests/features/steps/Steps.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# from behave import *
```
Comment (Aidan23Smith) [inline tests/features/steps/Steps.py:1]: Same as above

File: tests/features/steps/sign_in_steps.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+from behave import *
```
Comment (Aidan23Smith) [inline tests/features/steps/sign_in_steps.py:1]: Or this one

File: tests/features/user_journey.feature
Comment (Aidan23Smith) [inline tests/features/user_journey.feature:1]: Or this one

File: tests/unittests/Your_preferred_jobs_page_rendering_test.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+class MyTestCase(unittest.TestCase):
+    def setUp(self):
+        context = Context()
+        before_all(context)
+        context.browser.get('http://127.0.0.1:5000/preferred')
+        self.context = context
+        self.elements = Elements(context)
+
+        # self.pref_job_error =
```
Comment (Aidan23Smith) [inline tests/unittests/Your_preferred_jobs_page_rendering_test.py:20]: Remove this

File: tests/unittests/Your_preferred_jobs_page_rendering_test.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+    def test_pref_job_input_not_checked_when_no_3(self):
+        self.generic_elements()
+        self.elements.radio.click()
+        self.pref_job.send_keys('️')
+        self.elements.radio_2.click()
+        self.assertFalse(self.pref_job.is_displayed())
+        self.assertFalse(self.pref_job_label.is_displayed())
+        self.elements.button.click()
+        self.assertFalse(is_displayed(self.context.browser, 'pref_job-error'))
```
Comment (Aidan23Smith) [inline tests/unittests/Your_preferred_jobs_page_rendering_test.py:90]: Can parameterise these

File: requirements_dev.in (hunk 1 above)
Comment (nepnep83) [reply requirements_dev.in:10]: done

File: tests/features/features.feature (hunk 2 above)
Comment (nepnep83) [reply tests/features/features.feature:13]: The whole features file is now gone

File: tests/features/steps/Steps.py (hunk 3 above)
Comment (nepnep83) [reply tests/features/steps/Steps.py:1]: The whole features file is now gone

File: tests/features/steps/sign_in_steps.py (hunk 4 above)
Comment (nepnep83) [reply tests/features/steps/sign_in_steps.py:1]: The whole features file is now gone

File: tests/features/user_journey.feature
Comment (nepnep83) [reply tests/features/user_journey.feature:1]: The whole features file is now gone

File: tests/unittests/Your_preferred_jobs_page_rendering_test.py (hunk 6 above)
Comment (nepnep83) [reply tests/unittests/Your_preferred_jobs_page_rendering_test.py:90]: Yes

File: tests/unittests/Your_preferred_jobs_page_rendering_test.py (hunk 5 above)
Comment (nepnep83) [reply tests/unittests/Your_preferred_jobs_page_rendering_test.py:20]: Okay

**Review by Aidan23Smith — APPROVED** [review]

**Pull request merged**